src/test/other/contango/ContangoWand.t.sol

```diff
@@ -0,0 +1,616 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity >=0.8.13;
+
+import "forge-std/src/Test.sol";
+import "../../utils/TestConstants.sol";
+import "../../utils/Mocks.sol";
+
+import "../../../interfaces/IWitch.sol";
+import "../../../Cauldron.sol";
+import "../../../Join.sol";
+import "../../../other/contango/ContangoLadle.sol";
+import "../../../other/contango/ContangoWand.sol";
+
+contract ContangoWandTest is Test, TestConstants {
+    ICauldron internal contangoCauldron = ICauldron(0x44386ddB4C44E7CB8981f97AF89E928Ddd4258DD);
+    ILadle public immutable contangoLadle = ILadle(0x93343C08e2055b7793a3336d659Be348FC1B08f9);
+    IWitch internal immutable contangoWitch = IWitch(0x89343a24a217172A569A0bD68763Bf0671A3efd8);
+
+    ICauldron internal yieldCauldron = ICauldron(0x23cc87FBEBDD67ccE167Fa9Ec6Ad3b7fE3892E30);
+    ILadle public immutable yieldLadle = ILadle(0x16E25cf364CeCC305590128335B8f327975d0560);
+    address internal immutable yieldTimelock = 0xd0a22827Aed2eF5198EbEc0093EA33A4CD641b6c;
+
+    YieldSpaceMultiOracle public immutable yieldSpaceOracle =
+        YieldSpaceMultiOracle(0xb958bA862D70C0a4bD0ea976f9a1907686dd41e2);
+    CompositeMultiOracle public immutable compositeOracle =
+        CompositeMultiOracle(0x750B3a18115fe090Bc621F9E4B90bd442bcd02F2);
+
+    address internal bob = address(0xb0b);
+
+    ContangoWand internal wand;
+
+    function setUp() public virtual {
+        vm.createSelectFork("ARBITRUM", 65404751);
+
+        wand = new ContangoWand(
+            contangoCauldron,
+            yieldCauldron,
+            contangoLadle,
+            yieldLadle,
+            yieldSpaceOracle,
+            compositeOracle,
+            yieldTimelock,
+            contangoWitch
+        );
+
+        bytes4 root = 0x0;
+        wand.grantRole(root, addresses[ARBITRUM][TIMELOCK]);
+
+        vm.startPrank(addresses[ARBITRUM][TIMELOCK]);
+
+        AccessControl(address(yieldLadle.joins(USDT))).grantRole(root, address(wand));
+        AccessControl(address(yieldCauldron.series(FYUSDT2306).fyToken)).grantRole(root, address(wand));
+
+        wand.grantRole(wand.copySpotOracle.selector, address(this));
```

Comment:
You know that there is a `grantRoles` function, right?

Reply:
yes, but creating the array is not as pretty as I'd like, and this way I can group the related permissions (wand + cauldron or whatever)